### PR TITLE
feat(fxa-setting): Create IconListItem component

### PR DESCRIPTION
**Pull request by vpomerleau**
## Because

* We will be using the same list style in the new Recovery Key flow as in the PasswordInfoBalloon, and we want to make sure the styling is standardized for reuse.

## This pull request

* Create a new IconListItem component, with variations for key, shield, folder, globe, printer, and lock icons.
* Use this new component in the PasswordInfoBalloon

## Issue that this pull request solves

Closes: #FXA-7221

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Icons added:
image

In RTL:
image

With longer text, in AppLayout:
image

Updated PasswordInfoBalloon (no visual change)
image

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-settings/src/components/PasswordInfoBalloon/index.tsx [hunk 1]
```diff
@@ -4,16 +4,13 @@
...
   return (
-    <>
-      <div className="input-balloon flex gap-2 items-start text-xs">
-        <div className="ltr:mr-1 rtl:ml-1">
-          <IconKey className="w-5 h-5" aria-hidden="true" />
-        </div>
-        <div className="flex flex-col gap-4">
+    <ul className="input-balloon">
+      <KeyIconListItem>
+        <>
```
[the remark this answers is not in the document]
Comment (vpomerleau) [reply packages/fxa-settings/src/components/PasswordInfoBalloon/index.tsx:13]: There are two children, it's just hidden under the fold :)

File: packages/fxa-settings/src/components/IconListItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import { ReactElement } from 'react-markdown/lib/react-markdown';
+
+interface IconListItemProps {
+  icon: ReactElement;
+  children: string | ReactElement;
+}
+
+export const IconListItem = ({ icon, children }: IconListItemProps) => {
+  return (
+    <li className={`flex gap-2 items-start text-xs my-2`}>
```
[the remark this answers is not in the document]
Comment (vpomerleau) [reply packages/fxa-settings/src/components/IconListItem/index.tsx:21]: Oh yes! I initially had a `classnames` prop in here but decided it wasn't needed. I'll clean this up :)

File: packages/fxa-settings/src/components/PasswordInfoBalloon/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (millsoper) [reply packages/fxa-settings/src/components/PasswordInfoBalloon/index.tsx:13]: Oh gosh sorry !

**Review by millsoper — APPROVED** [review]
This looks great! 👍

**Comment by vpomerleau** [comment]
Added styling tweak (set svg path fill to "currentColor") to improve render with Windows high contrast mode.

**Pull request merged**